spotify_test.go

```diff
@@ -101,3 +103,58 @@ func TestNewReleasesRateLimitExceeded(t *testing.T) {
 		t.Error("Invalid data:", releases.Albums[0].ID)
 	}
 }
+
+func TestClient_Token(t *testing.T) {
+	// oauth setup for valid test token
+	config := oauth2.Config{
+		ClientID:     "test_client",
+		ClientSecret: "test_client_secret",
+		Endpoint:     oauth2.Endpoint{},
+		RedirectURL:  "http://redirect.redirect",
+		Scopes:       nil,
+	}
+	token := &oauth2.Token{
+		AccessToken:  "access_token",
+		TokenType:    "test_type",
+		RefreshToken: "refresh_token",
+		Expiry:       time.Now().Add(time.Hour),
+	}
+
+	// create and test client with valid test token
+	httpClient := config.Client(context.Background(), token)
+	client := New(httpClient)
+	token, err := client.Token()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if !token.Valid() {
+		t.Errorf("Token should be valid: %v", token)
+	}
+	if token.AccessToken != "access_token" {
+		t.Errorf("Invalid access token data: %s", token.AccessToken)
+	}
+	if token.RefreshToken != "refresh_token" {
+		t.Errorf("Invalid refresh token data: %s", token.RefreshToken)
+	}
+	if token.TokenType != "test_type" {
+		t.Errorf("Invalid token type: %s", token.TokenType)
+	}
+
+	// create and test client without an oauth token
```

Comment:
Where possible, break seperate cases out into subtests (https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks)

So in this case, you'd have a subtest for "success", "non oauth2 transport" and "invalid token"

It makes it a bit clearer what each case in the test is :)

Reply:
Yep. This is a nice way to test a function for all test cases I haven't heard of it yet.
I hope I fixed it correctly.